packages/sanity/src/core/form/studio/tree-editing/components/TreeEditingDialog.tsx

```diff
@@ -145,7 +146,12 @@ export function TreeEditingDialog(props: TreeEditingDialogProps): JSX.Element |
     [debouncedBuildTreeEditingState, onPathOpen],
   )
 
+  const shouldUseLegacyArrayDialog = useLegacyArrayEditingConfig()
+
   useEffect(() => {
+    // Check if the config has the legacy flag
+    if (shouldUseLegacyArrayDialog) return
```

Comment:
Do we need this check here since we are conditionally rendering the `TreeEditingDialog` based on this value in `FormBuilder`?